cloud/cloud_env.cc

```diff
@@ -59,6 +60,23 @@ void BucketOptions::SetBucketName(const std::string& bucket,
   }
 }
 
+void BucketOptions::SetObjectPath(const std::string& object) {
```

Comment:
Why is this needed?

Reply:
(I ask because we can likely offload this to filesystem library)

Reply:
The object path is the "directory" under which files are stored in the cloud.  Before this change, we stored the input name -- which is typically a directory path -- as-is as the object path, but this fails with Windows-style paths.

Since this object path ends up as part of the URL for the bucket, the path must follow URL syntax.  So the object path cannot have "\" or ":" in it.  This method implements the work to "standardize" the path to something that follows URL syntax.

If there is a means to get something from the filesystem library, I am not sure what it is.

Reply:
Can we just require object path to be URL-compatible? It's pretty clear this method is used to configure the path in the cloud, why are we feeding it local directory path with `C:\\`? We should either take what the user gives us verbatim or return an error if the path is not supported. Transparently changing the value here is weird.

Reply:
If we do not transform it, then this method must return a Status to say that it is malformed, which is doable.  The places that initialize the object path from dbname will break on Windows.  We would need to provide another function potentially to do the transformation.

If that is what you prefer, then it can be done and I will make the corresponding changes.

Reply:
> The places that initialize the object path from dbname will break on Windows.

Where does that happen? Is it only in tests? If it's only in tests we can have a function that transforms the path before calling `SetObjectPath`

> If that is what you prefer, then it can be done and I will make the corresponding changes.

Yeah that plan sounds good.